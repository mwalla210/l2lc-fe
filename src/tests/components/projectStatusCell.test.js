import React from 'react'
import renderer from 'react-test-renderer'
import ProjectStatusCell from '../../components/projectStatusCell'

const defaultOptions = {
  row:{
    value:'Completed'
  }
}

const alternateOptions = {
  row:{
    value:'On Hold'
  }
}

const tertiaryOptions = {
  row:{
    value:''
  }
}

describe('ProjectStatusCell', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <ProjectStatusCell {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it ('Renders with snapshot, alternate arguements', () => {
    let options = Object.assign({}, alternateOptions)
    const component = renderer.create(
      <ProjectStatusCell {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it ('Renders with snapshot, tertiary arguements', () => {
    let options = Object.assign({}, tertiaryOptions)
    const component = renderer.create(
      <ProjectStatusCell {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
