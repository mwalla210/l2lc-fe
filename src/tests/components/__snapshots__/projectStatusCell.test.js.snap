// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectStatusCell Renders with snapshot 1`] = `
<span>
  <span
    style={
      Object {
        "color": "#49a4ff",
        "transition": "all .3s ease",
      }
    }
  >
    ●
  </span>
   Completed
</span>
`;

exports[`ProjectStatusCell Renders with snapshot, alternate arguements 1`] = `
<span>
  <span
    style={
      Object {
        "color": "#ffbf00",
        "transition": "all .3s ease",
      }
    }
  >
    ●
  </span>
   On Hold
</span>
`;

exports[`ProjectStatusCell Renders with snapshot, tertiary arguements 1`] = `
<span>
  <span
    style={
      Object {
        "color": "#57d500",
        "transition": "all .3s ease",
      }
    }
  >
    ●
  </span>
   
</span>
`;
